editoast/src/views/v2/path/pathfinding.rs

```diff
@@ -186,27 +191,28 @@ async fn pathfinding_blocks_batch(
     Ok(pathfinding_results.into_iter().flatten().collect())
 }
 
-async fn build_pathfinding_request(
+fn build_pathfinding_request(
     pathfinding_input: &PathfindingInput,
-    conn: &mut DbConnection,
     infra: &Infra,
-) -> Result<std::result::Result<PathfindingRequest, PathfindingResult>> {
-    let path_items = &pathfinding_input.path_items;
+    path_item_cache: &PathItemCache,
+) -> std::result::Result<PathfindingRequest, PathfindingResult> {
+    let path_items: Vec<_> = pathfinding_input.path_items.iter().collect();
     if path_items.len() <= 1 {
-        return Ok(Err(PathfindingResult::NotEnoughPathItems));
+        return Err(PathfindingResult::NotEnoughPathItems);
     }
-    let track_offsets = match extract_location_from_path_items(conn, infra.id, path_items).await? {
-        Ok(track_offsets) => track_offsets,
-        Err(e) => return Ok(Err(e.into())),
-    };
+    let track_offsets = extract_location_from_path_items(&path_items, path_item_cache)?;
 
     // Check if tracks exist
-    if let Err(err) = check_tracks_from_path_items(conn, infra.id, &track_offsets).await? {
-        return Ok(Err(err));
+    for tracks in &track_offsets {
+        for track_offset in tracks {
+            if !path_item_cache.is_track_valid(&track_offset.track.0) {
+                return Err(PathfindingResult::NotFoundInTracks);
+            }
+        }
     }
```

Comment:
This feels like it belongs in `extract_location_from_path_items`
If I'm not mistaken, it's no use checking that operational point's tracks do exist, since the infra would be invalid anyway.
This check could then be performed in the `PathItemLocation::TrackOffset` branch of the mentionned function.

Reply:
Actually, it's checking that operational point's tracks do exist. The infra would be invalid but I this way we trigger a clean error.

Reply:
I feel like the responsibility of checking whether operational point's track sections exist does not belong to this function. 
Maybe the track locations in the `PathItemCache` could be filtered to contain only existing tracks for each operational point ?

Reply:
It will no longer be possible to differentiate between an `InvalidPathItem` and a `NotFoundInTracks`.

Reply:
> I feel like the responsibility of checking whether operational point's track sections exist does not belong to this function. 

I agree but I don't know how to move things around to make it more readable.